spyderlib/plugins/editor.py

```diff
@@ -492,7 +478,49 @@ def __init__(self, parent, ignore_last_opened_files=False):
             self.add_cursor_position_to_history(filename, position)
         self.update_cursorpos_actions()
         self.set_path()
-        
+
+    def set_open_filenames(self):
+        """ """
+        editorstack = self.editorstacks[0]
+        if self.projectexplorer:
+            if not self.projectexplorer.get_active_project():
+                filenames = []
+                filenames += [finfo.filename for finfo in editorstack.data]
+                self.set_option('filenames', filenames)
+
+    def setup_open_files(self):
+        """ """
+        active_project_path = None
+        if self.projectexplorer:
+             active_project_path = self.projectexplorer.get_active_project_path()
+
+        spyder_filenames = self.get_option('filenames', [])
+        project_filenames = self.projectexplorer.get_project_filenames()
+        filenames = spyder_filenames[:]
+
+        if active_project_path:
+            close_filenames = set(spyder_filenames) - set(project_filenames)
+        else:
+            close_filenames = set(project_filenames) - set(spyder_filenames)
+
+        filenames += project_filenames
+
+        if filenames:
+            self.load(filenames)
+            layout = self.get_option('layout_settings', None)
+            if layout is not None:
+                self.editorsplitter.set_layout_settings(layout)
+            win_layout = self.get_option('windows_layout_settings', None)
+            if win_layout:
+                for layout_settings in win_layout:
+                    self.editorwindows_to_be_created.append(layout_settings)
+            self.set_last_focus_editorstack(self, self.editorstacks[0])
+        else:
+            self.__load_temp_file()
+
+        for filename in close_filenames:
+            self.close_file_from_name(filename)

```

Comment:
Yes, let's do that but in code, i.e. not adding an option to our Preferences.

That way we can control when we want to open or not an empty file, and avoid all the workarounds you seem the be taken here :-)